examples/Metadata Extraction/Metadata-extraction.ps1

```diff
@@ -0,0 +1,287 @@
+        }
+        else
+        {
+            $RestoreAnalyticsClientEnablementState = "--no-enable-analytics-client"
+        }
+
+        "$( box configure:settings $RestoreAnalyticsClientEnablementState --analytics-client-name=$($this.OriginalAnalyticsClientName) )"
+        Write-Log "Restored original analytics client settings, name: $( $this.OriginalAnalyticsClientName ), enabled: $( $this.IsOriginalAnalyticsClientEnabled )." -output false
+    }
+
+    [void]
+    SetScriptAnalyticsClient()
+    {
+        if ( $this.IsAnalyticsClientSupported())
+        {
+            $this.StoreOriginalSettings()
+
+            "$( box configure:settings --enable-analytics-client --analytics-client-name=$($this.TemporaryAnalyticsClientName) )"
+            Write-Log "Set temporarily analytics client settings, name: $( $this.TemporaryAnalyticsClientName ), enabled: true." -output false
+        }
+    }
+
+    [void]
+    UnsetScriptAnalyticsClients()
+    {
+        if ( $this.IsAnalyticsClientSupported())
+        {
+            $this.RestoreOriginalSettings()
+        }
+    }
+}
+
+#############################################################################
+# Prompt for params if some are missing
+if (-not$folderID)
+{
+    Write-Log "Please enter the ID of the folder to extract metadata from:" -output true -color Yellow
+    $folderID = Read-Host
+}
+
+if (-not$UserId)
+{
+    Write-Log "Please specify the user ID  to pull metadata as a specific user" -output true -color Yellow
+    Write-Log "Press Enter if you want to run script as the current user." -output true -color Yellow
+    $UserId = Read-Host "User ID"
+}
+
+Function Start-Metadata-Extraction {
+
+    Write-Log "Pulling data from Folder ID: $folderID" -output true -color Green
+
+    try {
+        #Pull all File ID values from Folder ID
+        If (!$UserId) {
+            #Run as default user
+            Write-Log "No user ID specified. Using current user." -output true -color Yellow
+            $EntriesResp = (box folders:items $folderID --json 2>&1)
+        }
+        Else {
+            Write-Log "Extracting metadata as user ID: $UserId" -output true -color Yellow
+            #Run with as-user header with inputted User ID
+            $EntriesResp = (box folders:items $folderID --as-user=$UserId --json 2>&1)
+
+        }
+
+        $Entries = $EntriesResp | ConvertFrom-Json
+
+    }
+    catch {
+        Write-Log "Could not get the folder items. See error log for details." -errorMessage $EntriesResp -output True -color Red
+        break
+    }
+
+    Write-Log "Output $Entries"
+
+    ForEach ($Item in $Entries) {
+        $ItemID = $Item.id
+        Write-Log "Reading Item ID: $ItemID" -output true -color Green
+
+        #Pull Metadata values from Folder ID's
+        try {
+            If (!$UserId) {
+                #Run as default user (service account)
+                $MetadataResp = (box files:metadata $ItemID --json 2>&1)
```

Comment:
`$Item` can be File, Folder or Weblink also, it will return 404 if the item type not match.

Reply:
updated that part. good catch